R/content.R

```diff
@@ -1573,3 +1576,45 @@ search_content <- function(
 
   client$GET(path, query = query)
 }
+
+#' Convert content list to a data frame
+#'
+#' @description
+#' Converts a list returned by [search_content()] into a data frame.
+#'
+#' @param x A `connect_content_list` object (from [search_content()]).
+#' @param row.names Passed to [base::as.data.frame()].
+#' @param optional Passed to [base::as.data.frame()].
+#' @param ... Passed to [base::as.data.frame()].
+#'
+#' @return A `data.frame` with one row per content item.
+#' @export
+as.data.frame.connect_content_list <- function(
```

Comment:
How would this method differ for any other list-of-records from the API? This makes me think you want to have some inheritance, set the S3 class as something like `c("connect_content_list", "connect_list")`, and then you define the method on connect_list so any list-type container would be supported.